src/pages/merit/professors/[professorId].tsx

```diff
@@ -128,13 +137,29 @@ const ProfessorScoringPage: React.FC<ProfessorScoringPageProps> = ({
     return <ErrorMessage message={error} />;
   }
 
+  if (!professorId) {
+    return <ErrorMessage message="Missing Professor ID" />;
+  }
+
   const updateActivity = (updatedActivity: UpdateActivityDto) => {
     updateActivityClient(updatedActivity).then((res) => {
       if (res === ResponseStatus.Success) {
         setActivities((activities) => [
           ...activities.filter((u) => u.id !== updatedActivity.id),
           updatedActivity as ActivityDto,
         ]);
+
+        updateComputedProfessorScoreForUser(professorId).then((res) => {
+          if (res === ResponseStatus.Unauthorized) {
```

Comment:
this is super tiny but we have a utility function `isResponseError` that makes it much easier to handle these enumerated errors:
```
if (isErrorResponse(res)) setError(responseStatusMessage[res]);
```

check out the [ProfessorInfoForm](https://github.com/sandboxnu/faculty-activity-tracker/blob/2ae4e1e9a41bd18845149788e5fc8ed4c4f58f10/src/components/AccountSetup/ProfessorInfoForm.tsx#L138) for an example

Reply:
Right! I tried using that but then I realized that `res` is of the type `res: CreateProfessorScoreDto | ResponseStatus.UnknownError | ResponseStatus.Unauthorized | ResponseStatus.BadRequest`. As a result, I can't pass that into `isResponseError` because `CreateProfessorScoreDto` is also a variant.

I'll keep it as is for now till I can figure out a cleaner way of doing this!

Reply:
@RishikeshNK ahh yep I see that now. that's a flaw with how I wrote `isResponseError`: I meant to allow it to distinguish whether _any_ object is an error, not whether a certain ResponseStatus is an error. If you change the signature to this:
```
const isErrorResponse = (status: ResponseStatus | any): boolean
```
then we can use isErrorResponse to distinguish the `ResponseStatus` from union types